lib/completion.js

```diff
@@ -19,8 +19,19 @@ module.exports = function (yargs, usage) {
     // to completion().
     if (completionFunction) {
       if (completionFunction.length < 3) {
+        var result = completionFunction(current, argv)
+
+        // promise based completion function.
+        if (typeof result.then === 'function') {
+          return result.then(function (list) {
+            process.nextTick(function () { done(list) })
```

Comment:
Yes, it makes sure that if `done(list)` would throw an error, it wouldn't get swallowed by the promise and never reported.